chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -603,6 +614,20 @@ object RoomSpek : Spek({
             assertThat(pusherino.rooms).contains(pusherino.generalRoom)
         }
 
+        it ("joins room with other members") {
+            setUpInstanceWith(
+                    createDefaultRole(),
+                    newUsers(PUSHERINO, ALICE),
+                    newRoom(GENERAL, PUSHERINO, ALICE))
+
+            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
```

Comment:
Super user may see more data than a normal user and it is less likely it will be connecting via mobile SDK. The test could be simplified and brought closer to more common real scenario if it creates a room with one user and then another user joins that room.

Reply:
That's fair - i had tried to keep the method as similar to the first `joins room` test. I think I'll update the other `joins room` test to read the same as the current `val pusherino = chatFor(SUPER_USER).connect().assumeSuccess()` is misleading 😄

Reply:
The updating the `joins rooms` tests actually ended up being a bigger piece of work than I expected. The previous one worked as it was signing in as the super user which meant it was already part of the room, so not actually testing anything useful.

I've refactored but due to the nature of the joinedRoom method, I needed to get the general room Id which is why I sign in with the super user. The alternative way of doing this would have been to get the list of joinable rooms and then use that. Do you have a preference?

Reply:
I added setting room id as well for `newRoom` test setup (also fixing missing upgrade, all consulted with @mdpye), that allowed to simplify the test, please consider applying this patch:
```
Index: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/test/InstanceSupervisor.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/test/InstanceSupervisor.kt	(revision 9c4ddb75b5f2d1a47070162ce4746cbfdaf33493)
+++ chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/test/InstanceSupervisor.kt	(date 1580316928381)
@@ -87,7 +87,7 @@
     Instance(
             locator = INSTANCE_LOCATOR,
             serviceName = "chatkit",
-            serviceVersion = "v5",
+            serviceVersion = "v7",
             dependencies = TestDependencies()
     )
 }
@@ -307,6 +307,7 @@
                         path = "/rooms",
                         method = "POST",
                         body = mutableMapOf<String, Any?>(
+                                "id" to name,
                                 "name" to name,
                                 "push_notification_title_override" to pushNotificationTitleOverride,
                                 "user_ids" to userNames,
Index: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt	(revision 9c4ddb75b5f2d1a47070162ce4746cbfdaf33493)
+++ chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt	(date 1580316983035)
@@ -25,6 +25,7 @@
 import org.jetbrains.spek.api.Spek
 import org.jetbrains.spek.api.dsl.describe
 import org.jetbrains.spek.api.dsl.it
+import org.jetbrains.spek.api.dsl.xit
 import java.util.concurrent.ConcurrentLinkedQueue
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
@@ -649,11 +650,7 @@
                 }
             }.assumeSuccess()
 
-            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
-
-            val generalRoomId = superUser.rooms[0].id
-
-            val room = pusherino.joinRoom(generalRoomId).assumeSuccess()
+            val room = pusherino.joinRoom(GENERAL).assumeSuccess()
 
             assertThat(room.memberUserIds)
                     .containsExactly(ALICE, PUSHERINO, SUPER_USER)

```